### PR TITLE
setElement

### PR DESCRIPTION
**Pull request by gburnett**
Hi,

I'm using cordova-config and found that a setElement method was useful. 

Do you think this would be worth including?

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Hi @gburnett, thanks for the PR.

Can you add documentation for the `setElement` method? I guess we can use this one in the other methods to reduce code duplication.

Another thing that should be possible is passing the name of the tag as first parameter and the attributes as second attribute (so with no text).

``` js
config.setElement('author', {name: 'Foo Bar', email: 'foo@bar.com'});
```

**Comment by gburnett** [comment]
Ok, I'll work on that later today.

**Comment by gburnett** [comment]
Hi @SamVerschueren 

All ok here?

**Comment by SamVerschueren** [comment]
Thanks for the ping, will look at it shortly!

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 language: node_js
 node_js:
   - 'iojs'
+  - '0.10'
   - '0.12'
   - '4.0'
 after_script:
```
Comment (SamVerschueren) [inline .travis.yml:6]: Can you use this list?

```
- 'stable'
- '0.12'
- '0.10'
```

File: README.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ config.writeSync();
 
 #### file
 
-*Required*  
+*Required*
```
Comment (SamVerschueren) [inline README.md]: Add back the two spaces. It's probably the editor that removed them automatically.

File: README.md [hunk 3]
```diff
@@ -47,18 +47,41 @@ The path to the `config.xml` file.
 
 #### name
 
-*Required*  
+*Required*
```
Comment (SamVerschueren) [inline README.md]: Same here. And all the other `*Required*` lines in the file :).

File: README.md [hunk 4]
```diff
@@ -47,18 +47,41 @@ The path to the `config.xml` file.
...
 Type: `string`
 
 The name of the application.
 
+### #setElement(tag [, text [, attribs]])
```
Comment (SamVerschueren) [inline README.md]: #setElement(tag, [text], [attribs])

File: README.md [hunk 5]
```diff
@@ -47,18 +47,41 @@ The path to the `config.xml` file.
...
+
+#### text
+
+Type: `string`
+
+The element text.
+
+#### attribs
+
+Type: object
```
Comment (SamVerschueren) [inline README.md]: Type: `object`

File: index.js [hunk 6]
```diff
@@ -75,19 +75,47 @@ module.exports = (function() {
...
+
+            // Add the tag to the root
+            this._root.append(elementTag);
+        }
+
+        if (text) {
+            // set the text of the tag
+            elementTag.text = text;
+        } else {
+            elementTag.text = '';
```
Comment (SamVerschueren) [inline index.js]: Remove the `if` and use `elementTag.text = text || '';`.

File: index.js [hunk 7]
```diff
@@ -75,19 +75,47 @@ module.exports = (function() {
...
         }
 
-        // Set the text of the tag
-        nameTag.text = name;
+        elementTag.attrib = {};
```
Comment (SamVerschueren) [inline index.js]: Is this line necessary?

**Comment by SamVerschueren** [comment]
Hi @gburnett 
Sorry for the delay. Code looks much cleaner with this method so thank you very much! If you could have a look at the feedback I gave I would be very happy to merge this in :).

**Comment by gburnett** [comment]
Hi @SamVerschueren 

Thanks, I've made most of the corrections from your feedback.

For setElement I was following the example from setAuthor.  Should these really be different?

**Comment by SamVerschueren** [comment]
Thanks for the changes. The difference is that `attribs` can exist without `text`. This is not the case with `setAuthor`. That's why the notation should be different.

**Comment by gburnett** [comment]
Ok, cool.  I've updated that too.

**Comment by SamVerschueren** [comment]
Thanks for all the contributions! Appreciate it!

**Pull request merged**